xonsh/aliases.py

```diff
@@ -73,6 +73,19 @@ def source_bash(args, stdin=None):
     return
 
 
+def exec(args, stdin=None):
+    """
+    Replaces current process with command specified and passes in the
+    current xonsh environment.
+    """
+    env = builtins.__xonsh_env__
+    denv = env.detype()
+    try:
+        os.execvpe(args[0], args[1:len(args)], denv)
```

Comment:
You are doing it this way for the process replacement, rather than using subprocess, right?